src/phylo/nodes.jl

```diff
@@ -0,0 +1,655 @@
+@Docile.doc """
+PhyNode represents a node in a phylogenetic tree.
+
+A node can have:
+
+- `name`
+- `branchlength`
+- a reference to its `parent` PhyNode
+- reference to one or more `children`
+
+""" ->
+type PhyNode
+  name::String
+  branchlength::Float64
+  confidence::Float64
+  children::Vector{PhyNode}
+  parent::PhyNode
+
+"""
+Create a PhyNode.
+
+PhyNodes represent nodes in a phylogenetic tree. All arguments are optional 
+when creating PhyNodes:
+
+**Example:**
+
+    one = PhyNode()
+    two = PhyNode(name = "two",
+      branchlength = 1.0,
+      parent = one)
+
+**Parameters:**
+
+* `name`:         The name of the node (optional). Defaults to an empty string, indicating
+                  the node has no name.
+
+* `branchlength`: The branch length of the node from its parent (optional).
+                  Defaults to `-1.0`, indicating an unknown branch length.
+
+* `ext`:          An array of zero or more PhyExtensions (optional). Defaults to an empty
+                  array, i.e. `[]`, indicating there are no extensions.
+
+* `parent`:       The parent node (optional). Defaults to a self-reference, indicating
+                  the node has no parent.
+"""
+  function PhyNode(name::String = "", branchlength::Float64 = -1.0, confidence::Float64 = -1.0, children::Vector{PhyNode} = PhyNode[], parent = nothing)
```

Comment:
I worry a little about using `-1` to represent no value. Negative branch lengths or confidence values may not make sense, but I could imagine they arise anyway in some computation. What about using `Nullable{Float64}`?